### PR TITLE
mask keys by default, show them if desired

### LINKED ISSUE #52: Only expose URL explicitly
The URL always contains the API key and we should prohibit leaking the key accidentally. Therefore we should only return the URL if the user requests it explicitly and only in an interactive session.

- [x] remove url from results output
- [x] make urlonly option only available in interactive mode (via #35)
- [x] document the risk of exposing URL

### LINKED ISSUE #74: warn re API key with output = json_list, add_request = TRUE
`stopifnot(!interactive())` (or equivalent) if `output = "json_list"` and `add_request = TRUE`

### PR DESCRIPTION
**Pull request by dpprdan**
This PR masks the OpenCage API key by default when debugging with `url_only` and in `json_list` output when `add_request = TRUE`. 

The vignettes have been adjusted accordingly.

Unrelatedly, this also updates the code of conduct, following the [dev guide](https://devguide.ropensci.org/collaboration.html#coc-file).

### CODE REVIEW

File: vignettes/intro_opencage.Rmd [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: "Introduction to opencage"
 subtitle: "Forward and Reverse Geocoding"
 author: "Daniel Possenriede, Jesse Sadler, Maëlle Salmon"
-date: "2020-10-31"
+date: "2020-11-04"
```
Comment (maelle) [inline vignettes/intro_opencage.Rmd:5]: should this be an automatic date or no date?

**Review by maelle — APPROVED** [review]
nice work!

File: R/oc_config.R [hunk 2]
```diff
@@ -7,10 +7,17 @@
...
 #'   requests sent to the OpenCage API per second. Defaults to the value set in
 #'   the `oc_rate_sec` option, or, in case that does not exist, to 1L.
-#' @param no_record Logical vector of length one. When `TRUE` OpenCage will not
+#' @param no_record Logical vector of length one. When `TRUE`, OpenCage will not
 #'   create log entries of the queries and will not cache the geocoding
 #'   requests. Defaults to the value set in the `oc_no_record` option, or, in
-#'   case that does not exist, to FALSE.
+#'   case that does not exist, to `FALSE`.
+#' @param show_key Logical vector of length one. This is only relevant when
+#'   debugging `oc_forward()`  or `oc_reverse()` calls with the `return =
```
Comment (jessesadler) [inline R/oc_config.R]:
```suggestion
#'   debugging `oc_forward()` or `oc_reverse()` calls with the `return =
```

**Review by jessesadler — APPROVED** [review]
Nice work. These changes worked well when I tried them out.

File: vignettes/intro_opencage.Rmd (hunk 1 above)
Comment (dpprdan) [reply vignettes/intro_opencage.Rmd:5]: This is the date of when the vignette.Rmd was rendered from vignette.Rmd.orig. So it is kind of automatic, I guess? The rendering could probably be done by GitHub Actions, I guess?

In any case I'd prefer to have a date to easily see when the latest changes occurred (apart from having to look into the repo), I think.

Comment (maelle) [reply vignettes/intro_opencage.Rmd:5]: aaah right then it's fine

**Pull request merged**